src/flask_app.py

```diff
@@ -47,4 +48,9 @@ def flask_environment(self) -> dict[str, str]:
         secret_key_env = f"{FLASK_ENV_CONFIG_PREFIX}SECRET_KEY"
         if secret_key_env not in env:
             env[secret_key_env] = self._charm_state.flask_secret_key
+        for proxy_variable in ("http_proxy", "https_proxy", "no_proxy"):
```

Comment:
Would be possible/useful to have the same approach as used by GitHub Runner? Since is a common situation I think would be nice to follow a pattern for that.
https://github.com/canonical/github-runner-operator/blob/main/src/charm.py#L136

Reply:
Indeed that's also a nice approach, which is an alternative to moving these into CharmState as suggested in one of my comments

Reply:
For context, the GitHub runners don't implement the charm state pattern until we refactor it

The approach is ok, it stores the proxies on the charm. It would be better to have them on the state in general since modules might require them and should just have access to them via the state rather than needing something stored on the charm. The proxies are conceptually similar to things like database connection details and so on which should go to the state

Reply:
The challenge with storing proxy variables in the state is that they are only used once in the flask-k8s charm, in contrast to the GitHub runner charm, where they are used multiple times.

Reply:
What is the challenge with that?

Reply:
Incorporating proxies into the charm state requires the introduction of a new attribute to the charm state interface. However, given its limited utility - being used just once - I think it might not worth it.

Reply:
@gregory-schiano I think we should make a general decision here. I can see the argument - the proxy settings in this case just need to be mapped to a different environment variable.

The general case is that it might be required in a few places, e.g., written to the application config, written into the env etc. In general it seems to best go to the state

Reply:
@gregory-schiano and I discussed this as a part of another conversation, the proxy should go into the charm state

Reply:
Okay, done. The proxy information now exists in the charm state.